iparkerhodge/Terralanes
# Invitation, Sign In, and Forgot Password

### Updates
- replaced firebase authentication with AWS Cognito
- restyled login page to match mockups
- setup initial invitation email in Cognito console to send the username and temporary password
- enabled change password prompt on user's first sign in
- created forgot password form
- set up user identity verification email from Cognito system
- added verification code input that prompts new password form
- allowed password reset upon successful verification

### Steps to Test
1. Run the application locally on your machine
2. In the AWS Cognito User Pool console, create a new user with:
    - Username: {your email address}
    - Send an invitation to this user? should be checked. Email should be checked and SMS should be unchecked
    - Password: {any temporary password}
    - Leave phone number blank and uncheck "Mark phone number as verified?"
    - Email: {same as username}. Ensure "Mark email as verified?" is checked.
3. Go to your email inbox and ensure you received the invitation email. The link in the invitation email takes you to the deployed branch on AWS Amplify. For now, just test the application locally.
4. Login with the email and temporary password you provided. You should then be prompted to create a new password. Check the form validation to ensure that passwords match. Once submitted, you should be logged in.
5. Logout and click on the forgot password link. Input your email to receive a verification code.
6. Retrieve the verification code and input the code where prompted.
7. Create a new password and ensure the password reset is successful.
8. Login with the new password.
